### DISCUSSION TITLE
Screen capture ?

### DISCUSSION THREAD

**Discussion by myhillman** [dl9rdz/rdz_ttgo_sonde#425] [TTGO firmware]
Is there a way to take a screen capture ? If not, it would be a useful thing to add.

I have an ILI 9225 display, and I was thinking of documenting the displays for the wiki. I'm reluctant to resort to actual photos because of their quality, and I'd prefer actual image files. Could there be an endpoint on the internal server that captures the current screen?

**Comment by hr-ru** [comment]
Thanks, that is an interesting idea.  There is one issue, which is the limited RAM, the internal RAM of the ESP32 is just 512k (and there is below 100K free memory at the moment - parts of which is needed temporarily for handling web server requests). The ILI9225 full display content 176x220x3(rgb) would be around 110k RAM.  One would need to display some of the current features to make that fit.
Also, the ILI9225 display boards usually have just a one-directional interface, so it is not feasible to read the data out of the display either.

So it is not trivial.  There might be one option: Some (e.g. T-Beam does) but not all (e.g. the small TTGO Lora32 boards from Lilygo or Heltec don't) ESP32 boards have an external PSRAM chip that could be used.  It might be a feasible option to keep a copy of the display content in there, and make it accessible via the web interface.

**Reply by myhillman** [reply]
Shame - a great idea torpedoed by practical realities :(  Thanks for looking at it in detail.

**Comment by myhillman** [comment]
Just a thought - whilst rendering the screen, could you also render it in SVG to local ram for access via the web server. SVG is very small, and understood by all browsers. I know this is not trivial, but it would work no matter the display.

**Comment by Medvidas** [comment]
That idea is pretty cool. ILI LCD is rendering in RGB565 = ≈77 KB RAM needed, possible to much.
But wouldn't it be possible to render in parallel the data which are sended to LCD, 
convert them to 8bit pallete and compress line by line with lighweigth GIF routine?
